### ISSUE TITLE
Media playback - Content streaming vs. preloading

### ISSUE THREAD

**Issue by concreted** [Igalia/cog#135]
When playing back `webm` files, it appears cog is downloading the entire file first before starting playback. Is there any way to make it stream the video similar to desktop browsers e.g. Chrome? 

For example `cog -P fdo "https://upload.wikimedia.org/wikipedia/commons/3/3c/After_Storm_%284K_Resolution%29.webm"` - the playback starts very quickly on my laptop Chrome browser, but with cog it loads for a much longer time before starting playback. 

On larger videos the preloading seems to cause the device to run out memory as it tries to load the whole file into memory. `cog -P fdo "https://mira-public-assets.s3-us-west-2.amazonaws.com/4k-ultrahd.webm"` loads for a while before segfaulting on my 2GB RAM device since the file is ~2GB. Observing `top` I see the memory usage increasing until all memory is used.

**Comment by philn** [comment]
Can you check this also happens with wpewebkit 2.25.2?

**Comment by philn** [comment]
> On larger videos the preloading seems to cause the device to run out memory as it tries to load the whole file into memory. cog -P fdo "https://mira-public-assets.s3-us-west-2.amazonaws.com/4k-ultrahd.webm" loads for a while before segfaulting on my 2GB RAM device since the file is ~2GB. Observing top I see the memory usage increasing until all memory is used.

I can't reproduce this issue, top reports a stable mem usage for the WebProcess.

@telkamp if you have a test-case reproducing this issue feel free to share the details.

**Comment by telkamp** [comment]
Thank you for your response! 
After some further investigation, we found that "our" problem might be a bit different than the original issue #135: It seems that it is necessary to send thet HTTP fields in an exact order and/or to omit specific HTTP fields. But this is currently not verified.

Meanwhile. a colleague of mine has created new issue here: [#463](https://github.com/Igalia/cog/issues/463)
We will continue this issue #135 when we have more information.
